### ISSUE TITLE
Parsing issues in `list_tab` (salt/modules/cron.py)

### ISSUE THREAD

**Issue by swalladge** [saltstack/salt#38449]
The `list_tab(user)` function in `salt/modules/cron.py` has some issues parsing some values in the crontab. This causes some interesting bugs:

---

Problem: replaces sequences of spaces in commands with single spaces (see
state example below

Expected: command not modified

```
test spaces:
  cron.present:
    - user: me
    - name: 'echo "must  be  double  spaced"'
    - minute: 11
```

---

Problem: adds a new line every time the below state is run

Expected: detects already present the second time run

```
test env var with spaces:
  cron.env_present:
    - user: me
    - name: TEST_VAR
    - value: '"a string with plenty of spaces"'
```

---

### Versions Report

Salt Version:
           Salt: 2016.11.1
 
Dependency Versions:
           cffi: 1.9.1
       cherrypy: 8.1.2
       dateutil: 2.6.0
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.24.0
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov  7 2016, 11:55:55)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.2.0
 
System Versions:
           dist:   
        machine: x86_64
        release: 4.8.13-1-ARCH
         system: Linux
        version: Not Installed

**Comment by gtmanfred** [comment]
I believe I have fixed both of your issues with the following pull request.

https://github.com/saltstack/salt/pull/38487

**Comment by swalladge** [comment]
@gtmanfred please see comments on pull request.

Also I have begun work on a new implementation of the cron module at https://github.com/swalladge/cron_ng. I'm developing it as a custom module to begin with (and so I can test it quicker), but plan on integrating it with salt with tests and such and submitting a pull request to replace the current cron module if anyone is interested.

**Comment by cachedout** [comment]
Closed via #38487
